src/NServiceBus.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -98,50 +97,65 @@ string DefaultMessageIdStrategy(BasicDeliverEventArgs message)
 
         static string ValueToString(object value)
         {
-            var s = value as string;
-            if (s != null)
-            {
-                return s;
-            }
-
             var bytes = value as byte[];
             if (bytes != null)
             {
                 return Encoding.UTF8.GetString(bytes);
             }
 
-            var dictionary = value as IDictionary<string, object>;
+            var dictionary = value as Dictionary<string, object>;
             if (dictionary != null)
             {
-                var valuesToJoin = new List<string>();
+                var sb = new StringBuilder();
 
-                // ReSharper disable once LoopCanBeConvertedToQuery
                 foreach (var kvp in dictionary)
                 {
-                    valuesToJoin.Add(string.Concat(kvp.Key, "=", ValueToString(kvp.Value)));
+                    sb.Append(kvp.Key);
+                    sb.Append("=");
+                    sb.Append(ValueToString(kvp.Value));
+                    sb.Append(",");
                 }
 
-                return string.Join(",", valuesToJoin);
+                if (sb.Length > 0)
+                {
+                    sb.Remove(sb.Length - 1, 1);
+                }
+
+                return sb.ToString();
             }
```

Comment:
While I prefer this version of handling the `Dictionary<string, object>` case, it doesn't seem to provide any functional change. If I revert to the old method, the tests still pass.

Is this just a performance and/or readability optimisation? If so, I'd have preferred to see it in a separate refactoring commit, but I'm OK to leave it as-is in this case.